Shared/Flows/Onboarding/OnboardingViewController.swift

```diff
@@ -112,19 +96,11 @@ class OnboardingViewController: SwitchableContentViewController<OnboardingConten
                 }
 
                 guard let current = User.current() else { return }
-                if current.isOnboarded, current.status == .active {
-                    #if APPCLIP
-                    self.current = .waitlist(self.waitlistVC)
-                    #else
-                    self.showLoading(user: current)
-                    #endif
-                } else if current.status == .inactive, current.isOnboarded {
-                    #if APPCLIP
-                    self.current = .waitlist(self.waitlistVC)
-                    #else
-                    self.showLoading(user: current)
-                    #endif
-                } else {
+
+                switch current.status {
+                case .active:
+                    self.delegate.onboardingViewController(self, didOnboard: current)
+                case .needsVerification, .waitlist, .inactive, .none:
```

Comment:
When the code entry is complete, we either go to the name field, or finish onboarding if the user is activated.